packages/core/src/lib/readCsv/index.test.ts

```diff
@@ -0,0 +1,116 @@
+import { describe, it, expect } from 'vitest'
+import { syncReadCsv } from './index'
+
+describe('syncReadCsv', () => {
+  it('can parse JSON data', async () => {
+    const data = `
+name,payload
+John Doe,{"username":"johndoe","email":"johndoe@example.com","age":28,"gender":"male"}
+Alice Johnson,{"username":"alicejohnson","email":"alicej@example.com","age":25,"gender":"female"}
+`
+    const parsedCsvResult = await syncReadCsv(data, {
+      delimiter: ',',
+    })
+    const expectedCols = {
+      columns: [{ name: 'name' }, { name: 'payload' }],
+      bytes: expect.any(Number),
+      comment_lines: expect.any(Number),
+      empty_lines: expect.any(Number),
+      error: undefined,
+      header: false,
+      index: expect.any(Number),
+      invalid_field_length: expect.any(Number),
+      lines: expect.any(Number),
+      raw: undefined,
+      records: expect.any(Number),
+    }
+    expect(parsedCsvResult.value).toEqual({
+      rowCount: 2,
+      headers: ['name', 'payload'],
+      data: [
+        {
+          record: {
+            name: 'John Doe',
+            payload: {
+              username: 'johndoe',
+              email: 'johndoe@example.com',
+              age: 28,
+              gender: 'male',
+            },
+          },
+          info: expectedCols,
+        },
+        {
+          record: {
+            name: 'Alice Johnson',
+            payload: {
+              username: 'alicejohnson',
+              email: 'alicej@example.com',
+              age: 25,
+              gender: 'female',
+            },
+          },
+          info: expectedCols,
+        },
+      ],
+    })
+  })
```

Comment:
i would add some extra test cases maybe you can tell ai to generate some complex jsons with strings that have html escaping and what not, just to test it's robust.